core/src/test/java/org/elasticsearch/indices/state/OpenCloseIndexIT.java

```diff
@@ -190,6 +190,41 @@ public void testCloseOpenAllWildcard() {
         assertThat(openIndexResponse.isAcknowledged(), equalTo(true));
         assertIndexIsOpened("test1", "test2", "test3");
     }
+    
+    // if there are no indices to open/close throw an exception
+    public void testOpenCloseWildCardsNoIndicesDefault() {
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("_all").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("_all").execute().actionGet());
+    }
+    
+    // if there are no indices to open/close and allow_no_indices=true, the open/close is a no-op
+    public void testOpenCloseWildCardsNoIndicesAllowNoIndices() throws InterruptedException, ExecutionException {
+        IndicesOptions openIndicesOptions = IndicesOptions.fromOptions(false, true, false, true);
+        IndicesOptions closeIndicesOptions = IndicesOptions.fromOptions(false, true, true, false);
+
+        expectThrows(IndexNotFoundException.class,
+                () -> client().admin().indices().prepareOpen("test").setIndicesOptions(openIndicesOptions).execute().actionGet());
+        expectThrows(IndexNotFoundException.class,
+                () -> client().admin().indices().prepareClose("test").setIndicesOptions(closeIndicesOptions).execute().actionGet());
```

Comment:
@javanna Unfortunately it is not as straight forward as simply replacing `actionGet()` with `get()`. The difference between `get()` and `actionGet()` is that the later unwraps the `ExecutionException` and throws the actual cause (https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/ActionFuture.java#L37)

In my case when `get()` is used an `ExecutionException` is throws ( which wraps an `IndexNotFoundException` ). Checking for such a general exception is not really useful, so I will need to unwrap and check the actual cause. This will make a simple test more difficult to read and somewhat cumbersome. That's why I opted for the simpler use of `actionGet()` whenever an exception is expected and check the actual cause. And for the shorter version `get()` whenever no exception is expected.

Of course I can use `actionGet()` consistently for all the tests :)